scripts/time.js

```diff
@@ -1,16 +1,42 @@
-/*global $ apiUrl*/
+/*global $ apiUrl setInterval*/
 $(function () {
     "use strict";
     var timeSpan = $("#server-time");
+    var day = "";
+    var msec0 = Date.now();
 
-    function showTime(data) {
-        timeSpan.text(data.daytime);
+    function pad2(n) {
+        return ("0" + n).slice(-2);
     }
 
-    function updateClock() {
-        $.getJSON(apiUrl, {'q': 'v1/serverdaytime'}, showTime);
+    function showTime(hour, minute) {
+        timeSpan.text(day + " " + pad2(hour) + ":" + pad2(minute));
     }
 
-    setInterval(updateClock, 60000);
-    updateClock();
+    function resetTime(data) {
+        var hour = data.hour;
+        var minute = data.minute;
+        day = data.day;
+        // milliseconds from midnight
+        var msecs = 60000 * (parseInt(minute) + 60 * parseInt(hour));
+        // milliseconds from epoch to previous midnight
+        msec0 = Date.now() - msecs;
```

Comment:
Please don't try to do date math manually. Date math is hard and we should be using libraries that handle it and all of the edge cases for us. This won't account for daylight saving time changes.

The moment-based coding snippit from my prior comment should be all that you need to implement this without ever having to make an API call. It does rely on the client's local time & timezone being correct, but that is very reasonable. If that's not the level of precision that you are looking for, the server could load the page with the current time and the client could use moment to calculate a difference between it and the client time on page lod and use that, via moment, to calculate the time differences.